pkg/daemon/daemon.go

```diff
@@ -921,6 +921,13 @@ func (dn *Daemon) checkStateOnFirstRun() error {
 				return err
 			}
 		}
+		// if we're degraded here, it means we got an error likely on startup and we retried
+		// if it's the case, clear it out
+		if state.state == constants.MachineConfigDaemonStateDegraded {
+			if err := dn.nodeWriter.SetDone(dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, dn.name, state.currentConfig.GetName()); err != nil {
```

Comment:
Would it be better to just call `completeUpdate` as the code right above this is doing in case there's somehow still a cordon?  I wonder if we should always call `SetDone` here like this:

```
diff --git a/pkg/daemon/daemon.go b/pkg/daemon/daemon.go
index 0c8630d2..ff33afa0 100644
--- a/pkg/daemon/daemon.go
+++ b/pkg/daemon/daemon.go
@@ -895,9 +895,6 @@ func (dn *Daemon) checkStateOnFirstRun() error {
 		if dn.recorder != nil {
 			dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeNormal, "NodeDone", fmt.Sprintf("Setting node %s, currentConfig %s to Done", dn.node.Name, state.pendingConfig.GetName()))
 		}
-		if err := dn.nodeWriter.SetDone(dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, dn.name, state.pendingConfig.GetName()); err != nil {
-			return errors.Wrap(err, "error setting node's state to Done")
-		}
 		if out, err := dn.storePendingState(state.pendingConfig, 0); err != nil {
 			return errors.Wrapf(err, "failed to reset pending config: %s", string(out))
 		}
@@ -922,6 +919,11 @@ func (dn *Daemon) checkStateOnFirstRun() error {
 			}
 		}
 
+		if state.state != constants.MachineConfigDaemonStateDone {
+			if err := dn.nodeWriter.SetDone(dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, dn.name, state.pendingConfig.GetName()); err != nil {
+				return errors.Wrap(err, "error setting node's state to Done")
+			}
+		}
 		glog.Infof("In desired config %s", state.currentConfig.GetName())
 
 		// All good!
```

?

Reply:
Eh...it's OK we can do this type of detail in a later followup.